File: src/java/magmac/app/lang/node/PlantUMLHeader.java
package magmac.app.lang.node;

import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;
import magmac.app.compile.rule.ExactRule;
import magmac.app.compile.rule.TypeRule;

public class PlantUMLHeader implements PlantUMLRootSegment {
    static TypeRule createRule() {
        return new TypeRule("header", new ExactRule("@startuml\nskinparam linetype ortho"));
    }

    @Override
    public Node serialize() {
        return new MapNode("header");
    }
}

File: src/web/magmac/app/lang/node/PlantUMLHeader.ts
import { MapNode } from "../../../../magmac/app/compile/node/MapNode";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { ExactRule } from "../../../../magmac/app/compile/rule/ExactRule";
import { TypeRule } from "../../../../magmac/app/compile/rule/TypeRule";
export class PlantUMLHeader {
	createRule() : TypeRule {return new TypeRule( 0, new ExactRule( 0));;}
	serialize() : Node {return new MapNode( 0);;}
}
